src/Microsoft.ML.DataView/Microsoft.ML.DataView.csproj

```diff
@@ -5,7 +5,12 @@
     <IncludeInPackage>Microsoft.ML.DataView</IncludeInPackage>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <RestoreSources>$(RestoreSources);https://pkgs.dev.azure.com/foundry99/_packaging/SEAL/nuget/v3/index.json</RestoreSources>
+  </PropertyGroup>
+
   <ItemGroup>
+    <PackageReference Include="Microsoft.Research.SEALNet" Version="3.3.1" />
```

Comment:
We shouldn't be adding new dependencies on this project. People should be able to opt into optional functionality.

The beauty of the IDataView type system is that it is open/extensible. You can add `CiphertextDataViewType` in the `Microsoft.ML.SEAL` project instead. See the Onnx types, and the ImageDataViewType classes for examples.

Reply:
Unless something has changed since October 7th, the type system in not extensible in the manner needed for this extension. I have shown Zeeshan that I had to make these changes internally to get it to work. All of these changes were initially made in the Microsoft.ML.SEAL project and only moved to an internal project when visibility issues (largely the "internal" keyword on the vast majority of the ML.NET API) prevented access.

Reply:
> the type system in not extensible in the manner needed for this extension

We should sync on this then. If we need to add more extensibility mechanisms we can to solve this.